snap/README.md

```diff
@@ -21,37 +21,70 @@ The snap is published in the snap store at https://snapcraft.io/edgex-app-servic
 ```
 
 **Note** - the snap has only been tested on Ubuntu Core, Desktop, and Server.
 
 ## Using the EdgeX App Service Configurable snap
 
-The App Service Configurable application service allows a variety of use cases to be met by simply providing configuration (vs. writing code). For more information about this service, please refer to the README. As with device-mqtt, this service is disabled when first installed, as a profile must first be selected (see below) before the service is started. As with other EdgeX snaps, the `configuration.toml` files are found in the snap’s writable area:
+The App Service Configurable application service allows a variety of use cases to be met by simply providing configuration (vs. writing code). For more information about this service, please refer to the [README]
+(https://github.com/edgexfoundry/app-service-configurable/blob/main/README.md). As with device-mqtt, this service is disabled when first installed, as a profile must first be selected (see [profile](#profiles) section below) before the service is started. As with other EdgeX snaps, the `configuration.toml` files are found in the snap’s writable area: `/var/snap/edgex-app-service-configurable/current/config/res/`.
 
-/var/snap/edgex-app-service-configurable/current/config/res/
+### Snap configuration
+
+Application services implement a service dependency check on startup which ensures that all of the runtime dependencies of a particular service are met before the service transitions to an active state.
+
+Snapd doesn't support orchestration between services in different snaps. It is therefore possible on a reboot for an application service to come up faster than all of the required services running in the main edgexfoundry snap. If this happens, the application service may repeatedly fail startup, and if it exceeds the systemd default limits, then it might be left in a failed state. This situation might be more likely on constrained hardware (e.g. RPi).
+
+This snap, therefore, implements a basic retry loop with a maximum duration and sleep interval. If the dependent services are not available, the service sleeps for the defined interval (default: 1s) and then tries again up to a maximum duration (default: 60s). EdgeX services by default wait 60s for dependencies (e.g. Core Data) to become available, and will exit after this time if the dependencies aren't met. The following options can be used to override this startup behavior on systems where it takes longer than expected for the dependent services provided by the edgexfoundry snap to start. Note, both options below are specified as a number of seconds.
+    
+To change the maximum duration, use the following command:
+
+```bash
+sudo snap set edgex-app-service-configurable startup-duration=60
+```
+
+To change the interval between retries, use the following command:
+
+```bash
+sudo snap set edgex-app-service-configurable startup-interval=1
+```
```

Comment:
I believe this mechanism comes from the service and the snap only adds the config variables to set those environment variables.

https://docs.edgexfoundry.org/2.1/microservices/configuration/CommonEnvironmentVariables/

Why not add them to the Service Environment Configuration Overrides section?

Reply:
Updated.